be/src/exprs/agg/distinct.h

```diff
@@ -267,7 +265,8 @@ class TDistinctAggregateFunction
             if (slice.size > 16) {
                 mem_usage += this->data(state).deserialize_and_merge((const uint8_t*)slice.data, slice.size);
             } else {
-                T key = *reinterpret_cast<T*>(slice.data);
+                T key;
+                memcpy(&key, slice.data, sizeof(T));
```

Comment:
`reinterpret_cast` just copy point bytes to `T* data`, I think the reason is address of slice.data isn't aligned by 32 bytes. so... are you sure the modify can be resolve problem?

Reply:
It was because reinterpret_cast may violate C/C++ aliasing rules.

Reply:
We don't have any control over slice address. So for compiler, the right way should be:
1. if slice.data is 32 bytes aligned, use vmovdqa
2. if not, use vmovdqu.

But since we use `reinterpret_cast` to cast uin8_t* to a wider type int128_t*, it's undefined behavior.  And gcc compiler has the freedom to do some assumptions such as `slice.data must be 32bytes aligned`.

So for us, it's our responsibility to remove undefined behavior by using `memcpy`